monitoring/grafana/dashboards/tests/test_self.py

```diff
@@ -0,0 +1,35 @@
+from . import DashboardTest, PromqlTest
+
+
+@DashboardTest('hosts-overview.json')
+class TestSelf(PromqlTest):
+    """
+    Test PromqlTest logic
+    """
+
+    def test_fail(self):
+        self.set_expression('bonding_slaves > 0')
+
+        # input
+        self.add_series('bonding_slaves{master="bond0"}', '2')
+        self.add_series('bonding_slaves{master="bond1"}', '3')
+        self.add_series('bonding_slaves{master="bond2"}', '0')
+
+        # output
+        self.add_exp_samples('bonding_slaves{master="bond0"}', 2)
+
+        self.assertFalse(self.run_promtool(print_error=False))
```

Comment:
Not a big thing, but regardless of how the final YAML is generated (the `@dataclass` approach) I was thinking on exploring a [natural language approach](https://behave.readthedocs.io/en/stable/tutorial.html#feature-files):
```cucumber
@expected_failure
Scenario: "Test failure"
  Given the following series every "1m":
    | bonding_slaves{master="bond0"} | 2 |
    | bonding_slaves{master="bond1"} | 3 |
    | bonding_slaves{master="bond2"} | 0 |
  When "osd_hosts" is "127.0.0.1"
  Then panel "Network Load" shows:  # this code would search in all files for this expression
    | bonding_slaves{master="bond0"} | 2 |
```

The idea is that anyone can create and understand these (not only developers with Python expertise but also the users that often report and fix Grafana issues).

I thought about introducing behave (BDD testing) for the e2e UI testing, but this could also be interesting.